### DISCUSSION TITLE
getMessageIdFromTimestamp Timeout issues

### DISCUSSION THREAD

**Discussion by KannarFr** [apache/pulsar#22098] [Q&A] [answered]
Hi, when trying to getMessageIdFromTimestamp on a heavy topic with offloaded ledgers I get AdminProxyHandler Gateway Timeout after 30s. I am not sure to correctly identify the TO I should increase. Is that the idleTimeout in AdminProxyHandler ?

**Comment by lhotari** [comment]
This is not an answer to your question, but a comment about the underlying problem.  There's an issue https://github.com/apache/pulsar/issues/22129 requesting optimization of seeking by timestamp. There's an open PR https://github.com/apache/pulsar/pull/22152 to address this.

**Reply by KannarFr** [reply, marked as answer]
Thanks a lot for the pointer. I'm following it.

**Comment by lhotari** [comment]
@KannarFr There's no way to increase `idleTimeout` in AdminProxyHandler by configuration alone.  It would require code changes to address that. `idleTimeout` is documented in Jetty as "the max time, in milliseconds, a connection can be idle (that is, without traffic of bytes in either direction)". Feel free to create an issue and as always, contributions are welcome!

A practical maximum value for `idleTimeout` would be 59 seconds (59000ms) for environments where a NAT gateway or firewall has a timeout of 60 seconds. However, 30 seconds is already a very long time to wait for any response so increasing the limit isn't really a proper solution.
